src/collection/layout.js

```diff
@@ -74,6 +88,26 @@ var elesfn = ({
     } else {
       nodes.positions( fn );
 
+      if ( options.spacingFactor && options.spacingFactor !== 1){
```

Comment:
Could most of the logic in this block be refactored to a separate function to do the vector stuff.  That way, it can be reused for the top case and the bottom case, making things terser.